test/view_component/slotable_v2_test.rb

```diff
@@ -231,4 +231,24 @@ def test_renders_slots_with_before_render_hook
     assert_selector(".greeting", text: "Hello, John Doe")
     assert_selector(".greeting", text: "Hello, Jane Doe")
   end
+
+  def test_renders_slots_with_default_args
+    render_inline(SlotsV2WithDefaultArgsComponent.new) do |component|
+      component.icon
```

Comment:
I think this is failing because this isn't setting the value of `component.icon`, but accessing it.

To set a slot, you have to pass an argument or a block.